### PR TITLE
[rcanvas] remove RAttrBox, introduce RAttrBorder

### PR DESCRIPTION
**Pull request by linev**
`RAttrBox` had too many encapsulations. Use instead directly `RAttrFill` and newly introduced `RAttrBorder` for `RFrame`, `RBox`, `RPave`

Remove `RAttrOnFrame` class which was used in inheritance for the drawables, put "onframe" and "clipping" attributes directly in the classes where supported.

Simplify naming convention. For every attributes aggregation like `RAttrLine` provide only two methods:

     RAttrLine &AttrLine() { return fAttrLine; }
     const RAttrLine &AttrLine() const { return fAttrLine; }

Update JSROOT to reflect changes on C++ side

### CODE REVIEW

File: graf2d/gpadv7/inc/ROOT/RAttrAxis.hxx [hunk 1]
```diff
@@ -113,8 +113,7 @@ class RAttrAxis : public RAttrBase {
       fTimeFormat.Clear();
    }
 
-   const RAttrLine &GetAttrLine() const { return fAttrLine; }
-   RAttrAxis &SetAttrLine(const RAttrLine &line) { fAttrLine = line; return *this; }
+   const RAttrLine &AttrLine() const { return fAttrLine; }
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/inc/ROOT/RAttrAxis.hxx:116]: Given that we're already in "Attr land", would it be enough to call this:
```suggestion
   const RAttrLine &Line() const { return fAttrLine; }
```
?

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx [hunk 2]
```diff
@@ -6,39 +6,40 @@
...
-class RAttrOnFrame {
+class RAttrBorder : public RAttrLine {
 
-   RAttrValue<bool> fOnFrame;    ///<! is drawable drawn on the frame or not
-   RAttrValue<bool> fClipping;   ///<! if drawable should be clipped by frame
+   RAttrValue<int>     fRx{this, "rx", 0};              ///<! rounding on x coordinate, px
+   RAttrValue<int>     fRy{this, "ry", 0};              ///<! rounding on y coordinate, px
 
-public:
+   R__ATTR_CLASS(RAttrBorder, "border");
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:31]: Is this public? If so I'd appreciate an explicit
```suggestion
public:
   R__ATTR_CLASS(RAttrBorder, "border");
```

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx [hunk 3]
```diff
@@ -6,39 +6,40 @@
...
+   ///The rounding on x, px
+   RAttrBorder &SetRx(int rx) { fRx = rx; return *this; }
+   int GetRx() const { return fRx; }
 
-   void SetOnFrame(bool on = true) { fOnFrame = on; }
-   bool GetOnFrame() const { return fOnFrame; }
-
-   void SetClipping(bool on = true) { fClipping = on; }
-   bool GetClipping() const { return fClipping; }
+   ///The rounding on x, px
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:37]:
```suggestion
   ///The rounding on y, py
```

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx [hunk 4]
```diff
@@ -6,39 +6,40 @@
...
-   RAttrValue<bool> fOnFrame;    ///<! is drawable drawn on the frame or not
-   RAttrValue<bool> fClipping;   ///<! if drawable should be clipped by frame
+   RAttrValue<int>     fRx{this, "rx", 0};              ///<! rounding on x coordinate, px
+   RAttrValue<int>     fRy{this, "ry", 0};              ///<! rounding on y coordinate, px
 
-public:
+   R__ATTR_CLASS(RAttrBorder, "border");
 
-   RAttrOnFrame(RDrawable *drawable) : fOnFrame(drawable, "onframe", false), fClipping(drawable, "clipping", false) {}
+   ///The rounding on x, px
```
Comment (Axel-Naumann) [inline graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:33]: What is "px" (see also for y)?

**Review by Axel-Naumann — APPROVED** [review]
Nice!

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx (hunk 2 above)
Comment (linev) [reply graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:31]: Macro `R__ATTR_CLASS` has protected and public methods and define visibility scope itself

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx (hunk 3 above)
Comment (linev) [reply graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:37]: Here `px` means pixels

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx (hunk 4 above)
Comment (linev) [reply graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:33]: Also here `px` is pixels

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx (hunk 3 above)
Comment (Axel-Naumann) [reply graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:37]: OK, would be great if you can spell that you :-)

File: graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx (hunk 2 above)
Comment (Axel-Naumann) [reply graf2d/gpadv7/inc/ROOT/RAttrBorder.hxx:31]: OK but then we *must* have a subsequent `public:` or `private:` - we cannot rely on readers of the class to know what the access is after `R__ATTR_CLASS`. IIRC that was missing?

**Pull request merged**